framework_tvb/tvb/adapters/simulator/simulator_fragments.py

```diff
@@ -71,18 +71,32 @@ def fill_trait(self, datatype):
 
 
 class SimulatorRMFragment(ABCAdapterForm):
-    def __init__(self, prefix='', project_id=None, surface_index=None):
+    def __init__(self, prefix='', project_id=None, surface_index=None, connectivity_gid=None):
         super(SimulatorRMFragment, self).__init__(prefix, project_id)
         conditions = None
         if surface_index:
-            conditions = FilterChain(fields=[FilterChain.datatype + '.fk_surface_gid'], operations=["=="],
-                                     values=[str(surface_index.gid)])
+            conditions = FilterChain(fields=[FilterChain.datatype + '.fk_surface_gid',
+                                             FilterChain.datatype + '.fk_connectivity_gid'],
+                                     operations=["==", "=="],
+                                     values=[str(surface_index.gid), connectivity_gid.hex])
         self.rm = TraitDataTypeSelectField(CortexViewModel.region_mapping_data, self, name='region_mapping',
                                            conditions=conditions)
         self.lc = TraitDataTypeSelectField(CortexViewModel.local_connectivity, self, name='local_connectivity',
                                            conditions=conditions)
```

Comment:
I do not see field  `LocalConnectivityIndex. fk_connectivity_gid`. Will this filter ever work for LocalConnectivity?

Reply:
You are right, the connectivity filter should have been added only for the region mapping.